### PR TITLE
Implement Protector Aasimar Radiant Soul

### LINKED ISSUE #212: look into protector assimar
from discord 
> found another bug ish,, warlocks hexblade cures, and barbarian rage show up in beyond20 options, i now have protector aasimar - Your transformation lasts for 1 minute or until you end it as a bonus action. During it, you have a flying speed of 30 feet, and once on each of your turns, you can deal extra radiant damage to one target when you deal damage to it with an attack or a spell. The extra radiant damage equals your level. its nots showing a button to add damage.

### PR DESCRIPTION
**Pull request by Aeristoka**
Fixes #212

### CODE REVIEW

File: src/dndbeyond/content-scripts/character.js [hunk 1]
```diff
@@ -319,6 +319,13 @@ function rollItem(force_display = false) {
             damage_types.push("Psychic");
             character.mergeCharacterSettings({ "bard-psychic-blades": false });
         }
+        //Protector Aasimar: Radiant Soul Damage
+        if (character.hasRacialTrait("Radiant Soul") &&
+            character.getSetting("protector-aasimar-radiant-soul", true)) {
+            damages.push(" + " + character._level);
```
Comment (kakaroto) [inline src/dndbeyond/content-scripts/character.js:325]: nitpick but no need for the "+" here, it's an independent damage of '3', there's nothing before it, so it doesn't need to be an equation.

File: src/dndbeyond/content-scripts/character.js [hunk 2]
```diff
@@ -319,6 +319,13 @@ function rollItem(force_display = false) {
             damage_types.push("Psychic");
             character.mergeCharacterSettings({ "bard-psychic-blades": false });
         }
+        //Protector Aasimar: Radiant Soul Damage
+        if (character.hasRacialTrait("Radiant Soul") &&
+            character.getSetting("protector-aasimar-radiant-soul", true)) {
```
[the remark this answers is not in the document]
Comment (Aeristoka) [reply src/dndbeyond/content-scripts/character.js:324]: Fixed

File: src/common/settings.js [hunk 3]
```diff
@@ -436,6 +436,12 @@ const character_settings = {
         "description": "Use your Gathered Swarm ability to add extra Force damage to your weapon attacks",
         "type": "bool",
         "default": false
+    },
+    "protector-aasimar-radiant-soul": {
+        "title": "Protector Aasimar: Radiant Soul",
+        "description": "Once on each of your turns, while transformed, you can deal extra radiant damage to one target when you deal damage to it with an attack or a spell. The extra radiant damage equals your level.",
```
[the remark this answers is not in the document]
Comment (Aeristoka) [reply src/common/settings.js:442]: Changed to that text. I really did just write a shortened version of the rules before.

File: src/dndbeyond/content-scripts/character.js [hunk 4]
```diff
@@ -319,6 +319,13 @@ function rollItem(force_display = false) {
...
+            damages.push(" + " + character._level);
+            damage_types.push("Radiant Soul");
+            character.mergeCharacterSettings({ "protector-aasimar-radiant-soul": false });
```
[the remark this answers is not in the document]
Comment (Aeristoka) [reply src/dndbeyond/content-scripts/character.js:327]: Noted and changed.

File: src/dndbeyond/content-scripts/character.js (hunk 1 above)
Comment (Aeristoka) [reply src/dndbeyond/content-scripts/character.js:325]: Fixed. I had based it of Great Weapon Master and Sharpshooter, so should those be changed as well?

image

**Comment by Aeristoka** [comment]
Added Radiant Soul Damage possibility to rollAction and rollSpell (I had missed those before)

**Pull request merged**